core/types/transaction.go

```diff
@@ -459,28 +461,56 @@ func (s TxByNonce) Len() int           { return len(s) }
 func (s TxByNonce) Less(i, j int) bool { return s[i].Nonce() < s[j].Nonce() }
 func (s TxByNonce) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 
-type _Order interface {
+type OrderType int
+
+const (
+	Tx OrderType = iota
+	Bundle
+)
+
+type Order interface {
 	AsTx() *Transaction
-	AsBundle() *SimulatedBundle
+	AsBundle() *MevBundle
+	AsSimulatedBundle() *SimulatedBundle
+	TxType() OrderType
```

Comment:
I think the TxType is not needed, since it's sufficient to check the cast (it's the same complexity) unless you do switch statements (but I don't think you do)